pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -528,3 +542,37 @@ func makeLabels(s *v1alpha1.TaskRun) map[string]string {
 func isDone(status *v1alpha1.TaskRunStatus) bool {
 	return !status.GetCondition(duckv1alpha1.ConditionSucceeded).IsUnknown()
 }
+
+func (c *Reconciler) checkTimeout(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec) (bool, error) {
```

Comment:
if you pass the `Delete` function into here, you can move this function into a separate package and have unit tests for it without involving the reconciler :D (for example see [ResolveTaskResources](https://github.com/knative/build-pipeline/blob/master/pkg/reconciler/v1alpha1/taskrun/resources/taskresourceresolution.go#L39))